tests/testthat/test-shape.R

```diff
@@ -83,3 +83,23 @@ test_that("shape_broadcast_() applies recycling rules", {
     class = "vctrs_error_incompatible_type"
   )
 })
+
+test_that("can combine shaped native classes (#1290, #1329)", {
+  x <- Sys.time() + c(1, 1e6)
```

Comment:
Would you mind using `new_datetime(c(1, 1e6))`? `Sys.time()` makes me slightly nervous